File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  // Put your code here
  int squareOfSum(int n) {
    var sum = 0;

    for (var i = 1; i <= n; i++) {
      sum += i;
    }

    return pow(sum, 2) as int;
  }

  int sumOfSquares(int n) {
    var sum = 0;

    for (var i = 1; i <= n; i++) {
      sum += pow(i, 2) as int;
    }

    return sum;
  }

  int differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}

File: go/difference-of-squares/difference_of_squares.go
package differenceofsquares

func SquareOfSum(n int) int {
	s := (n * (n + 1) / 2)
	return s * s
}

func SumOfSquares(n int) int {
	s := 0
	for i := 1; i <= n; i++ {
		s += i * i
	}
	return s
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
